### PR TITLE
[react-select] Complete ControlProps

### PR DESCRIPTION
**Pull request by Methuselah96**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/JedWatson/react-select/blob/master/packages/react-select/src/Select.js#L1849-L1859 and https://github.com/JedWatson/react-select/blob/master/packages/react-select/src/components/Control.js#L61-L71>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

Requested here: https://github.com/DefinitelyTyped/DefinitelyTyped/commit/6a5fb3eb56c80448b114180c89a6d618c0104ae0#r41518733.

### CODE REVIEW

**Comment by Methuselah96** [comment]
@simon-asp Would you like to review this?

File: types/react-select/src/components/Control.d.ts [hunk 1]
```diff
@@ -1,28 +1,27 @@
-import { ComponentType, ReactNode, Ref as ElementRef } from 'react';
+import { ComponentType, ReactNode, RefCallback, MouseEventHandler, TouchEventHandler } from 'react';
 
 import { borderRadius, colors, spacing } from '../theme';
 import { CommonProps, OptionTypeBase, PropsWithStyles } from '../types';
```
Comment (santialbo) [inline types/react-select/src/components/Control.d.ts:4]:
```suggestion
import { CommonProps, OptionTypeBase, PropsWithStyles, Theme } from '../types';
```

File: types/react-select/src/components/Control.d.ts [hunk 2]
```diff
@@ -1,28 +1,27 @@
...
+      onTouchEnd: TouchEventHandler<HTMLDivElement>;
+    };
+    /** Whether the select is disabled. */
+    isDisabled: boolean;
+    /** Whether the select is focused. */
+    isFocused: boolean;
+    /** Whether the select is expanded. */
+    menuIsOpen: boolean;
+    /** Children to render. */
+    children: ReactNode;
```
Comment (santialbo) [inline types/react-select/src/components/Control.d.ts:21]: Would you mind also adding the theme here in this PR? 
```suggestion
    children: ReactNode;
    /** The style theme */
    theme: Theme;
```

**Review by santialbo — CHANGES_REQUESTED** [review]
Would you mind also adding the theme here in this PR?

File: types/react-select/src/components/Control.d.ts (hunk 2 above)
Comment (Methuselah96) [reply types/react-select/src/components/Control.d.ts:21]: The `theme` prop should go on the `CommonProps` interface not on the `ControlsProps` interface. I think it makes sense for that to be a separate PR.

**Pull request closed without merging**